runtime/parachains/src/scheduler.rs

```diff
@@ -214,28 +210,17 @@ impl<T: Config> Pallet<T> {
 				.for_each(|(freed_index, freed_reason)| {
 					match &cores[freed_index.0 as usize] {
 						CoreOccupied::Free => {},
-						CoreOccupied::Parachain(_) => {}, // If we ever do slot sharing parachains, this case needs to be handled
-						CoreOccupied::Parathread(entry) => {
-							match freed_reason {
-								FreedReason::Concluded => {
-									concluded_paras.insert(freed_index, entry.claim.0);
-								},
-								FreedReason::TimedOut => {
-									if entry.retries <
-										T::AssignmentProvider::get_max_retries(freed_index)
-									{
-										let entry = ParathreadEntry {
-											retries: entry.retries + 1,
-											claim: entry.claim.clone(),
-										};
-										timedout_paras
-											.insert(freed_index, Assignment::ParathreadA(entry));
-									} else {
-										// Consider max retried parathreads as concluded for the assignment provider
-										concluded_paras.insert(freed_index, entry.claim.0);
-									}
-								},
-							};
+						CoreOccupied::Paras(entry) => {
+							if <paras::Pallet<T>>::is_parathread(entry.para_id) {
```

Comment:
Because the old implementation doesn't handle parachains here at all. At first I didn't have the distinction here but that broke quite a few things. Worth revisiting in the end if/how to remove it, imho.

Reply:
When we push back, it goes to the front of the assignment provider. This means we should push in reverse order:

Last item in the claim queue, then the second last item, until the first, then the occupied core itself. Then we arrive at the same order as we popped:

1. Push: Last item
2. Push: Second last item | Last item
3. Push: First item | Second last item | Last item
4. Push Occupied core | First item | Second last item | Last item

With the current code we would arrive at:

Occupied core | Last item | Second last item | First item

Reply:
Some unit tests and the adder collator test. Though I just tried again and it worked just fine. I assume there was some other error as I was making changes.